examples/aliquot-sum.masm

```diff
@@ -0,0 +1,47 @@
+# x: input number
+# f: current factor candidate
+# a: aliquot sum
+
+begin                     # [x]
+  push.0 swap             # [x, a]
+  push.0                  # [f, x, a]
+  push.1
+  while.true
+    # increment f
+    add.1                 # [f, x, a]
+
+    # check whether f is a factor of x
+    dup.1 dup.1           # [f, x, f, x, a]
+    u32checked_mod        # [x % f, f, x, a]
+    push.0 u32checked_eq  # [x % f == 0, f, x, a]
+                          # [factor?, f, x, a]
+
+    # multiply by boolean trick to avoid expensive conditional
+    dup.1 mul             # if factor?:
+                          #   [f, f, x, a]
+                          # else:
+                          #   [0, f, x, a]
+
+    movup.3               # if factor?:
+                          #   [a, f, f, x]
+                          # else:
+                          #   [a, 0, f, x]
+
+    add                   # if factor?:
+                          #   [a + f, f, x]
+                          # else:
+                          #   [a, f, x]
+    # now the top of the stack has the new value of a
+
+    movdn.2               # [f, x, a]
```

Comment:
You're correct, but I think it's fine as it is, since `a+f`is the new value of `a`, and we'll need to update the terms at some point or the comments will get unwieldy.